### ISSUE TITLE
[feature] Take detected values for not explicitly defined settings (e.g. in package:compile.runtime...)

### ISSUE THREAD

**Issue by forry** [conan-io/conan#7951]
Hello,
in regards to #615, I have a situation where I want to use a different build of a package while using the default for others. The suggested proccess is to use -s package:'\<setting\>'. I think that this should only change the vaule of \<setting\> for the package while taking the default values for other settings for that package (and other packages too). If this assumption is correct and I have this settings:
```
[settings]
arch=x86_64
arch_build=x86_64
build_type=Release
compiler=Visual Studio
compiler.runtime=MD
compiler.version=16
os=Windows
os_build=Windows
testingPackage:compiler.runtime=MDd
testingPackage:build_type=Debug
```
Then this behavior of conan seems inconsistent - conan install (shortened) output:
'Please define compiler value first too'

I Would expect that the compiler would be taken as "Visual Studio" (and all other compiler.subvalues) and only the compiler.runtime gets changed. The problem is that we are using cmake and therefore a conan_run macro wrapped in another macro and multiple builds. So if for example server or some other automation is building a different version of a software with different settings.compiler.whatever (e.g toolset) and/or different compiler and the dev wants for all of them (which he doesn't know what they are) change only the runtime (for whatever reason) it seems to be unachievable or problematic.

I hope that my explanation is not confusing :).

Thanks,
forry

- [x] I've seen the [CONTRIBUTING guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).

**Comment by memsharded** [comment]
Hi @forry 

Yes, I have managed to reproduce this behavior. I am submitting a Pull Request #7953 to see if this can be changed, or maybe it might be breaking behavior and can't be done, let's see.

Thanks for reporting!

**Comment by forry** [comment]
Thank you very much for such a prompt response.

**Comment by memsharded** [comment]
https://github.com/conan-io/conan/pull/7953 has been merged, this will be released and work in next Conan 1.31 release.

Thanks for your feedback!
